src/main/java/com/statful/client/CustomMetric.java

```diff
@@ -115,6 +124,40 @@ public String toMetricLine() {
         return metricLineBuilder.build();
     }
 
+    private List<Aggregation> supplyGlobalAggregations() {
+        if (!getMetricType().isPresent()) {
+            return Collections.emptyList();
+        } else {
+            switch (getMetricType().get()) {
+                case GAUGE:
```

Comment:
I'm thinking if we could move this "switch" to the metric type enum.

The method would receive the options object and return the proper aggregation based on it's own type.

The same is valid for the frequencies

Reply:
Sounds good.